web/app/dashboard/dashboard.view.controller.js

```diff
@@ -63,4 +64,64 @@
     vm.toggleEdit = function() {
         $location.url("/edit/" + dashboard.id);
     };
+
+    vm.isListening = false;
+
+    vm.supportsSpeech = ('webkitSpeechRecognition' in window);
+    var speech = vm.supportsSpeech ? new webkitSpeechRecognition() : null;
+
+    vm.listen = function() {
+        if (!vm.supportsSpeech) {
+            console.error('No support for speech recognition on this platform!');
+            return;
+        }
+
+        var started_listening = 'Speak now...';
+        configureSpeech();
+        console.log(started_listening);
+        vm.speechOutput = started_listening;
+        vm.isListening = true;
+        speech.start();
+    }
+
+    vm.stopListening = function() {
+        speech.stop();   
+    }
+
+    function configureSpeech() {
+        speech.continuous = true;
+        speech.interimResults = true;
+        speech.lang = $rootScope.locale;
+
+        speech.onresult = function(e) {
+            var interim_transcript = '';
+            var final_transcript = '';
+
+            if (typeof(e.results) === 'undefined') {
+              return;
+            }
+            for (var i = e.resultIndex; i < e.results.length; ++i) {
+              var val = e.results[i][0].transcript;
+              if (e.results[i].isFinal) {
+                final_transcript += " " + val;
+              } else {
+                interim_transcript += " " + val;
+              }
+            }
+            vm.speechOutput = interim_transcript.trim();
+
+            if (final_transcript) {
+                vm.speechOutput = final_transcript.trim();
+                OHService.sendVoice(vm.speechOutput);
+                setTimeout(2000);
+                this.stop();
+            }
+        };
+
+        speech.onend = function(e) {
+            console.log('Stopped listening');
+            vm.speechOutput = '';
+            vm.isListening = false;
+        }
+    }
   }
```

Comment:
Yes, I even tried that but failed. Maybe you'd have a suggestion how to handle passing values to `vm.speechOutput` and `vm.isListening`?
And then there's this `OHService.sendVoice(vm.speechOutput);` which inside `speech.service.js` would result in circular reference (I guess).